healpy/fitsfunc.py

```diff
@@ -397,15 +398,21 @@ def read_map(
 
     for ff, curr_dtype in zip(field, dtype):
         try:
```

Comment:
Is the try/except statement statement still necessary? Perhaps we can simply call `hdu.verify("fix")`, and then proceed to read?

On a related note, it would also be good to gradually move away from print statements and use the `logging` module or (even better suited for users) `warnings.warn()`.

Reply:
I prefer to use the try/except, not sure how expensive is the verification step.

yes, definitely that would be great to remove all the prints, probably `warnings` is enough for our purposes